### PR TITLE
Secure bootstrap: Rewritten and improved

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/551/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/551/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-551/output/_site/

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/553/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/553/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-553/output/_site/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/554/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/554/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-554/output/_site/

**Review by nickanderson — APPROVED** [review]
Some suggestions/comments. but looks much better than it did previously.

File: getting-started/installation/secure-bootstrap.markdown [hunk 1]
```diff
@@ -5,116 +5,245 @@ published: true
...
-"bootstrap" procedure:
+In the default configuration, the policy server (`cf-serverd`) on the hub machine trusts incoming connections from the same `/16` subnet.
+This means that:
 
-`cf-agent --bootstrap $HUB_IP`
+* Bootstrapping new clients will work as long as the 2 first numbers in the IP address are identical ([IPv4 dot decimal representation](https://en.wikipedia.org/wiki/Dot-decimal_notation)) .
+  The hub and client mutually accept each other's keys, automatically.
+* This applies to _all_ IP addresses within that range, not just the 1 IP address belonging to the client you are currently bootstrapping.
+* The hub will keep accepting new clients from those IP addresses until you change the configuration.
+* If you try to bootstrap a client where those 2 numbers in the IP address do not match the hub, it will fail.
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply getting-started/installation/secure-bootstrap.markdown:31]: They are numbers.

File: getting-started/installation/secure-bootstrap.markdown [hunk 2]
```diff
@@ -5,116 +5,245 @@ published: true
...
+However, this is in the default configuration, and there are several limitations and implications of this;
+
+## Default configuration
 
-CFEngine's trust model is based on the secure exchange of keys. This
-exchange of keys between *client* and *hub*, can either happen manually
-or automatically. Usually this step is automated as a dead-simple
-"bootstrap" procedure:
+In the default configuration, the policy server (`cf-serverd`) on the hub machine trusts incoming connections from the same `/16` subnet.
+This means that:
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply getting-started/installation/secure-bootstrap.markdown:25]: Yea, no I don't want to do that here.

File: getting-started/installation/secure-bootstrap.markdown [hunk 3]
```diff
@@ -5,116 +5,245 @@ published: true
...
+In order to specify and limit which hosts (IP addresses) are considered trusted and allowed to connect and fetch policy files, you can put the trusted IP addresses and subnets into the `acl` variable:
 
-We must change the policy we're distributing to fully locked-down
-settings. So after we have set-up our hub (using the standard procedure
-of `cf-agent --bootstrap $HUB_IP`) we take care of the following:
+```json
+[file=/var/cfengine/masterfiles/def.json]
+{
+  "variables": {
+    "default:def.acl": ["1.2.3.4", "4.3.2.1"]
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply getting-started/installation/secure-bootstrap.markdown]: Fixed.

File: getting-started/installation/secure-bootstrap.markdown [hunk 4]
```diff
@@ -5,116 +5,245 @@ published: true
...
 
-However there are cases where initial CFEngine deployment is happening
-over an insecure network, for example the Internet. In such cases we
-already have a secure channel to the clients, usually ssh, and we use
-this channel to *manually establish trust* from the hub to the clients
-and vice-versa.
+## Allowing only specific IP addresses / subnets
 
-## Locking down the policy server
+In order to specify and limit which hosts (IP addresses) are considered trusted and allowed to connect and fetch policy files, you can put the trusted IP addresses and subnets into the `acl` variable:
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply getting-started/installation/secure-bootstrap.markdown]: Thanks, fixed.

**Comment by olehermanse** [comment]
@cf-bottom Jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/555/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/555/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-555/output/_site/

**Review by nickanderson — APPROVED** [review]

**Comment by olehermanse** [comment]
Cherry-pick: https://github.com/cfengine/documentation/pull/3249

**Review by nickanderson — APPROVED** [review]

**Pull request merged**